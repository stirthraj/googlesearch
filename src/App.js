import { useState } from "react";
import Search from "./components/Search";


function App() {
  const [c,setC]=useState("");

  return (
    <>
      <div className="main" style={{backgroundColor:c}}>
        <Search colorFunction={(color) => setC(color)} />
      </div>
    </>
  );
}

export default App;
