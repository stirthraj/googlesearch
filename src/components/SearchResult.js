function SearchResult(props){
    console.log("props:"+props);
    if(typeof(props.data)==="object"){
      const Results = props.data.map((res) => (
        <div key={res.id } className="card">
          <div style={{ margin: "10px" }}>
            <div className="card-link">{res.formattedUrl}</div>
            <a href={res.formattedUrl}>
              <h3
                className="card-title"
                dangerouslySetInnerHTML={{ __html: res.htmlTitle }}
              />
            </a>
            <div>{res.snippet}</div>
          </div>
        </div>
      ));
return (
  <>
    <h3>Search Results</h3>
    {Results}
  </>
);
    }
    else{
      return(<><div>
        Error to get request
        </div></>)
    }
    
}
export default SearchResult;

