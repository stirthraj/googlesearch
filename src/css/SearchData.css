*,*::after,*::before{
    box-sizing: border-box;
}
.Sticky{
    top: 0;
    display: flex;
    position: fixed;
    width: 100%;
    padding: 5px;
    background-color: #fff;
    box-shadow: 0 2px 8px 1px rgb(64 60 67 / 24%);
    border-color: rgba(223,225,229,0);
}
.Sticky-logo{
    height: 35px;
    width: 160px;
    padding: 4px 30px 0px 30px;
}
.Sticky-input{
    padding: 10px 100px 10px 20px;
    position: relative;
    border-radius: 10px;
    border: 1px solid #1670BE;
    box-shadow: 0 0 3px #1670BE;
}
@media only screen and (max-width: 600px){
  .Sticky-input{
    padding: 10px 50px 10px 50px;
    position: relative;
    border-radius: 10px;
    border: 1px solid #1670BE;
    box-shadow: 0 0 3px #1670BE;
}  
.Sticky{
    top: 0;
    display: flex;
    position: fixed;
    width: 100%;
    padding: 1px;
    background-color: #fff;
    box-shadow: 0 2px 8px 1px rgb(64 60 67 / 24%);
    border-color: rgba(223,225,229,0);
}
.Sticky-input{
    width: 100%;
    position: relative;
    border-radius: 10px;
    border: 1px solid #1670BE;
    box-shadow: 0 0 3px #1670BE;
}
.Sticky-logo{
    height: 35px;
    width: 100px;
    padding: 3px;
}
.Sign-up{
    display: none;
}
}
.Sticky-input:focus{
    border: 1px solid #1670BE;
    box-shadow: 0 0 3px #1670BE;
    outline-offset: 0px;
    outline: transparent;
}
.Sticky-bar{
    border-radius: 10px;
}
.Sticky-bar-mic{
position: absolute;
    margin-top: 8px;
    margin-left: -70px;
}
.Sticky-bar-search{
  margin-top: 6px;
  margin-left: 5px;
}
.Sticky-btn{
    background-color: transparent;
    margin-top: 1px;
    position: absolute;
    margin-left: -50px;
    border: transparent;
}
.Sign-up{
    margin-left: 7%;
}
.Sign-btn{
    background-color:#1a73e8;
    border:1px solid transparent;
    color:white;
    border-radius: 5px;
    padding-left: 20px;
    padding-right: 20px;
    padding-top: 10px;
    padding-bottom: 10px;
}

.container{
    width: 700px;
    margin: 100px 220px 20px 160px;
}
@media only screen and (max-width:600px){
    .container{
        width: 100%;
        margin: 5px;
    }
}
.card{
   padding: 0px 2px;
   margin: 0px 2px;
}
.card-title{
    margin: 2px 0px;
    padding: 2px 0px;
}
.Sticky + .container {
  padding-top: 42px;
}
a{
    text-decoration: none;
}
a:hover{
    text-decoration: underline;
}
.footer{
    background-color:#f2f2f2;
    padding: 10px;
    margin-bottom: 0px;
}
.footer>h4{
    text-align: center;
}
.card-link{
    color: grey;
}