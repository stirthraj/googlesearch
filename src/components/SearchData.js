import { useEffect, useRef, useState } from "react";
import "../css/SearchData.css";
import search from "../img/search.svg";
import mic from "../img/googlemic_clr_24px.svg";
import logo from "../img/google_logo.svg";
import SearchResult from "./SearchResult";

function SearchData(props) {
  const [query, setQuery] = useState(props.query);
  const [result, setResult] = useState([]);

  const qq = useRef();
  const onKeyUpHandle = (e) => {
    if (e.keyCode === 13) {
      e.preventDefault();
      setQuery(e.target.value);
    }
  };
  const API_KEY = "AIzaSyDrLZZoy4wQ9 - t0dJvWAfMxWlhfvdeXdak";
  const CONTEXT_KEY = "b285d02aa413dc4f4";

  useEffect(() => {
    async function GoogleQuery() {
      await fetch(
        `https://www.googleapis.com/customsearch/v1?key=${API_KEY}&cx=${CONTEXT_KEY}&q=${query}`
      )
        .then((res) => res.json())
        .then((res) => setResult(res.items))
        .catch((error) => {
          console.log(error);
        });
      document.title = `Result of ${query}`;
    }
    GoogleQuery();
  }, [query]);
  return (
    <>
      <div className="Sticky">
        <div>
          <img className="Sticky-logo" src={logo} alt="logo" />
        </div>
        <div className="Sticky-bar">
          <input
            ref={qq}
            className="Sticky-input"
            size="50"
            defaultValue={query}
            onKeyUp={onKeyUpHandle}
            placeholder="Search Google or type a URL"
          />
          <button
            className="Sticky-btn"
            onClick={() => setQuery(qq.current.value)}
          >
            <img className="Sticky-bar-search" src={search} alt="icon" />
          </button>
          <img className="Sticky-bar-mic" src={mic} alt="mic-icon" />
        </div>
        <div className="Sign-up">
          <button className="Sign-btn">Sign in</button>
        </div>
      </div>

      <div className="container">
        <SearchResult data={result} />
      </div>
      <p></p>
      <div className="footer">
        <h4>Eduwheel India &copy; @2021-2022</h4>
      </div>
    </>
  );
}

export default SearchData;
