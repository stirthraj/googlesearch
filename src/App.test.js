import { mount, shallow } from "enzyme";
import App from "./App";
import SearchData from "./components/SearchData";
import SearchResult from "./components/SearchResult";
import React from "react";
import renderer from "react-test-renderer";

describe("Testing",()=>{
  test("adds 1 + 2 to equal 3", () => {
    expect(1+2).toBe(3);
  });

  test("object testing", () => {
    const data = { one: 1 };
    data["two"] = 2;
    expect(data).toEqual({ one: 1, two: 2 });
  });

  test("adding positive numbers is not zero", () => {
    for (let a = 0; a < 10; a++) {
      for (let b = 1; b < 10; b++) {
        expect(a + b).not.toBe(0);
      }
    }
  });

  test("numbers null", () => {
    const n = null;
    expect(n).toBeNull();
    expect(n).toBeDefined();
    expect(n).not.toBeUndefined();
  });

  test("tobelessthanorequal testing", () => {
    const value = 2 + 2;
    expect(value).toBeLessThanOrEqual(4);
  });

  test("string match", () => {
    expect("tirth").toMatch(/[a-z]/);
    expect("trithraj").not.toMatch(/I/);
  });

  const data = ["tir", "tirth", "hii"];

  test("the shopping list has tirth on it", () => {
    expect(data).toContain("tir");
  });


// //for element
  it("api fetch without Error", () => {
    const wrapper = shallow(<SearchResult />);
    const welcome = <h3>Search Results</h3>;
    expect(wrapper.contains(welcome)).toEqual(false);
  });
//for component
  it("App renders without crashing", () => {
    shallow(<App />);
  });

  test("SearchResult renders without crashing",()=>{
    shallow(<SearchResult/>);
  })
    test("SearchData renders without crashing", () => {
      shallow(<SearchData/>);
    });


    it("testing props", () => {
      const wrapper = mount(<SearchData data={data} />);
      expect(wrapper.props().data).toEqual(data);
    });
    it("contains users account email", () => {
      const wrapper = mount(<SearchData data={data} />);
      const value = wrapper.find(".container").text();
      expect(value).not.toEqual("");
    });
    //state checking
it.skip("SearchData renders correctly with no error message", () => {
  const wrapper = mount(<SearchData/>);
  expect(wrapper.state("error")).toEqual(undefined);
});

//snapshot testing

it("Snapshot SearchResult correctly", () => {
  const tree = renderer
    .create(<SearchResult/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("Snapshot SearchData correctly", () => {
  const tree = renderer.create(<SearchData />).toJSON();
  expect(tree).toMatchSnapshot();
});



});