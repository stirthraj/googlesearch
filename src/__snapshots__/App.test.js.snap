// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Snapshot SearchData correctly 1`] = `
Array [
  <div
    className="Sticky"
  >
    <div>
      <img
        alt="logo"
        className="Sticky-logo"
        src="google_logo.svg"
      />
    </div>
    <div
      className="Sticky-bar"
    >
      <input
        className="Sticky-input"
        onKeyUp={[Function]}
        placeholder="Search Google or type a URL"
        size="50"
      />
      <button
        className="Sticky-btn"
        onClick={[Function]}
      >
        <img
          alt="icon"
          className="Sticky-bar-search"
          src="search.svg"
        />
      </button>
      <img
        alt="mic-icon"
        className="Sticky-bar-mic"
        src="googlemic_clr_24px.svg"
      />
    </div>
    <div
      className="Sign-up"
    >
      <button
        className="Sign-btn"
      >
        Sign in
      </button>
    </div>
  </div>,
  <div
    className="container"
  >
    <h3>
      Search Results
    </h3>
  </div>,
  <p />,
  <div
    className="footer"
  >
    <h4>
      KloudOne Â© @2021-2022
    </h4>
  </div>,
]
`;

exports[`Testing Snapshot SearchResult correctly 1`] = `
<div>
  Error to get request
</div>
`;
